internal/broker/broker.go

```diff
@@ -412,67 +442,103 @@ func (b *Broker) handleIsAuthenticated(ctx context.Context, session *sessionInfo
 	case "device_auth":
 		response, ok := session.authInfo["response"].(*oauth2.DeviceAuthResponse)
 		if !ok {
-			return AuthDenied, `{"message": "could not get required response"}`
+			return isAuthenticatedResult{
+				access:  AuthDenied,
+				Message: "could not get required response",
+			}
 		}
 
 		t, err := b.auth.oauthCfg.DeviceAccessToken(ctx, response, b.providerInfo.AuthOptions()...)
 		if err != nil {
-			return AuthRetry, fmt.Sprintf(`{"message": "could not authenticate user: %v"}`, err)
+			return isAuthenticatedResult{
+				access:  AuthRetry,
+				Message: fmt.Sprintf("could not authenticate user: %v", err),
+			}
 		}
 
 		rawIDToken, ok := t.Extra("id_token").(string)
 		if !ok {
-			return AuthDenied, `{"message": "could not get id_token"}`
+			return isAuthenticatedResult{
```

Comment:
This is not C or other languages. Go allows multiple returned values, with last one being an error by convention. Don’t create and wrap this to return a simple struct.